File: 백준/Silver/18258. 큐 2/큐 2.py
import sys
from collections import deque

que = deque()
N = int(sys.stdin.readline())
for _ in range(N) :
    i = sys.stdin.readline().split()

    if i[0] == 'push':
        que.append(int(i[1]))

    elif i[0] == 'pop':
        if que:
            print(que[0])
            que.popleft()
        else :
            print(-1)

    elif i[0] == 'size':
        print(len(que))

    elif i[0] == 'empty':
        if que:
            print (0)
        else :
            print (1)
            
    elif i[0] == 'front':
        if que:
            print(que[0])
        else:
            print(-1)
            
    elif i[0] == 'back':
        if que:
            print(que[-1])   
        else:
            print(-1)

File: 백준/Silver/18258. 큐 2/큐 2.cc
#include <iostream>
#include <queue>

using namespace std;

int main() {

	ios_base :: sync_with_stdio(false);
	cin.tie(NULL);

	queue<int> Q;
	string command;
	int t, cnt;

	cin >> t;

	for (int i = 0; i < t; i++) {
		cin >> command;
		
		if (command == "push") {
			cin >> cnt;
			Q.push(cnt);
		}
		else if (command== "pop") {
			if (Q.empty()) {
				cout << "-1" << '\n';
			} else {
				cout << Q.front() << '\n';
				Q.pop();
			}
		}
		else if (command == "front") {
			if (Q.empty()) {
				cout << "-1" << '\n';
			}
			else {
				cout << Q.front() << '\n';
			}
		}
		else if (command == "back") {
			if (Q.empty()) {
				cout << "-1" << '\n';
			}
			else {
				cout << Q.back() << '\n';
			}
		}
		else if (command == "size")
			cout << Q.size() << '\n';
		else if (command == "empty")
			cout << Q.empty() << '\n';
		else if (command == "top") {
			if (Q.empty()) {
				cout << "-1" << '\n';
			}
			else {
				cout << Q.front() << '\n';
			}
		}
	}

	

	return 0;
}